types/aws-lambda/trigger/api-gateway-proxy.d.ts

```diff
@@ -128,22 +129,29 @@ export interface APIGatewayProxyEventV2 {
         time: string;
         timeEpoch: number;
     };
-    body: string;
-    pathParameters: { [name: string]: string };
+    body?: string;
```

Comment:
Are the newly optional properties not provided only in the API tester? That's had the same sort of different behavior to the actual API before, and the existing payload effectively says that the tester is wrong (though mostly because of compatibility)

Reply:
Optional properties are not provided in real deployed API. For example, if cookies aren't sent with the http request, API Gateway v2 doesn't include `cookies` field in proxy lambda's event. I believe v1's event also needs to be checked for this.

Reply:
Unfortunately it's a bit late for V1 even if so, it would basically break every API lambda out there. An earlier PR to do so was reverted after (IMO valid) feedback. Possibly easier to stomach now with the conditional access operator though?